src/test/java/io/redis/examples/QueryEmExample.java

```diff
@@ -0,0 +1,347 @@
+        SearchResult res1 = jedis.ftSearch(
+            "idx:bicycle",
+            new Query("@price:[270 270]")
+        );
+        System.out.println(res1.getTotalResults()); // >>> 1
+
+        List<Document> docs1 = res1.getDocuments();
+
+        for (int i = 0; i < docs1.size(); i++) {
+            System.out.println(docs1.get(i).getId());
+        }
+        // >>> bicycle:0
+
+        SearchResult res2 = jedis.ftSearch(
+            "idx:bicycle",
+            new Query().addFilter(
+                new Query.NumericFilter("price", 270, 270)
+            )
+        );
+        System.out.println(res2.getTotalResults()); // >>> 1
+
+        List<Document> docs2 = res2.getDocuments();
+
+        for (int i = 0; i < docs2.size(); i++) {
+            System.out.println(docs2.get(i).getId());
+        }
+        // >>> bicycle:0
+        // STEP_END
+
+        // Tests for 'em1' step.
+        // REMOVE_START
+        Assert.assertEquals(1, res1.getTotalResults());
+        Assert.assertEquals("bicycle:0", docs1.get(0).getId());
+
+        Assert.assertEquals(1, res2.getTotalResults());
+        Assert.assertEquals("bicycle:0", docs2.get(0).getId());
+        // REMOVE_END
+
+
+        // STEP_START em2
+        SearchResult res3 = jedis.ftSearch(
+            "idx:bicycle",
+            new Query("@condition:{new}")
+        );
+        System.out.println(res3.getTotalResults()); // >>> 5
+
+        List<Document> docs3 = res3.getDocuments();
+
+        for (int i = 0; i < docs3.size(); i++) {
+            System.out.println(docs3.get(i).getId());
+        }
+        // >>> bicycle:5
+        // >>> bicycle:0
+        // >>> bicycle:6
+        // >>> bicycle:7
+        // >>> bicycle:8
+        // STEP_END
+
+        // Tests for 'em2' step.
+        // REMOVE_START
+        Assert.assertEquals(5, res3.getTotalResults());
+        Assert.assertEquals("bicycle:5", docs3.get(0).getId());
+        Assert.assertEquals("bicycle:0", docs3.get(1).getId());
+        Assert.assertEquals("bicycle:6", docs3.get(2).getId());
+        Assert.assertEquals("bicycle:7", docs3.get(3).getId());
+        Assert.assertEquals("bicycle:8", docs3.get(4).getId());
+        // REMOVE_END
+
+
+        // STEP_START em3
+        SchemaField[] emailSchema = {
+            TextField.of("$.email").as("email")
+        };
+
+        jedis.ftCreate("idx:email", 
+            new FTCreateParams()
+                .addPrefix("key:")
+                .on(IndexDataType.JSON),
+            emailSchema
+        );
+
+        jedis.jsonSet("key:1", new Path2("$"), "{\"email\": \"test@redis.com\"}");
+
+        try {
+            SearchResult res4 = jedis.ftSearch(
+                "idx:email",
+                "test@redis.com",
+                new FTSearchParams().dialect(2)
+            );
+            System.out.println(res4.getTotalResults());
+        } catch (JedisDataException j) {
+            System.out.println("Query syntax 'test@redis.com' not supported in Jedis.");
```

Comment:
Did you try helper methods in RediSearchUtil?

Reply:
@sazzad16 I didn't know about RediSearchUtil until now, but I've tried it and it works fine. Thanks for the advice!

Reply:
@andy-stark-redis No problem! Happy to help :slightly_smiling_face: